compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1159,6 +1159,11 @@ pub enum Rvalue<'tcx> {
     /// optimizations and codegen backends that previously had to handle deref operations anywhere
     /// in a place.
     CopyForDeref(Place<'tcx>),
+
+    /// Computes the offset of a field.
+    ///
+    /// This is generated by `offset_of!`
+    OffsetOf(Vec<Field>, Ty<'tcx>),
```

Comment:
This should be a variant of `NullaryOp`

Reply:
This makes `Rvalue` 8 bytes larger. I hope that's not a problem.

Reply:
It should probably be `&'tcx [Field]` or whatever the interned form of a list is, rather than `Vec`?

Reply:
That did be `rustc_middle::ty::list::List`.